WordPress/gutenberg
# Add border support to site logo

## What?

Add border block support to the site logo.
Partial for https://github.com/WordPress/gutenberg/issues/43247, https://github.com/WordPress/gutenberg/issues/37757

## Why?

To allow more customization of the block, and for consistency with other blocks.

## How?

The border is applied to the image element, not the wrapper.

## Testing Instructions
Please test both the placeholder and the site logo with image, with and without link, duotone, rounded style variation, and padding.
When testing with dutotone, you should find that the duotone changes the border color. This is consistent with other image blocks that supports both.
When testing padding, you should find that the padding is on the wrapper, which means it is not between the image and the border but between the border and other page content. This is also consistent with other blocks.

1. Activate a block theme that does not style the logo, for example, Twenty Twenty-Three.
2. Open the Site Editor.
3. Add a site logo block.
4. Next, open the Styles sidebar. Open the blocks panel and select the site logo.
5. Confirm that there is  a new button for opening the border styles panel. Open the panel and adjust the border.
If you are using a placeholder, you will see that the border has a low opacity, but it should be present.
I found that this is inconsistent across blocks. The image block placeholder has no visible border, the featured image has a fully visible border, not a border with low opacity.
Confirm that the border displays correctly when an image is selected.
6. Save and view the front. Confirm that the logo and border shows correctly.
7. Return to the site editor. With the block selected, open the block settings sidebar and locate the border panel.
8. Adjust the border. Please test both palette color and custom border color. The settings should override the border from the global styles.


## Screenshots or screencast 

<img width="1507" alt="global styles border option for site logo" src="https://user-images.githubusercontent.com/7422055/221100803-ca97bc6b-fd0b-43b4-91d9-a530f34fd984.png">
